From 295ac770c58534772a51ecdaf3d83205428a9009 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Tue, 30 Oct 2018 11:42:48 +0000
Subject: [PATCH 276/432] staging: vc-sm-cma: Fixup driver for older VCHI APIs

Original patch was based off staging which included some cleanups
of the VCHI APIs. Those aren't present here, so switch back to
the older API.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/staging/vc04_services/vc-sm-cma/vc_sm.c          | 2 +-
 drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
index b63285ff3b1f..a8758ab8c87f 100644
--- a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
+++ b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
@@ -632,7 +632,7 @@ static void vc_sm_connected_init(void)
 		goto err_free_mem;
 	}
 
-	ret = vchi_connect(vchi_instance);
+	ret = vchi_connect(NULL, 0, vchi_instance);
 	if (ret) {
 		pr_err("[%s]: failed to connect VCHI instance (ret=%d)\n",
 		       __func__, ret);
diff --git a/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c b/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
index 7a5c2fd76c45..cc5b31a54cda 100644
--- a/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
+++ b/drivers/staging/vc04_services/vc-sm-cma/vc_sm_cma_vchi.c
@@ -325,8 +325,13 @@ struct sm_instance *vc_sm_cma_vchi_init(VCHI_INSTANCE_T vchi_instance,
 		SERVICE_CREATION_T params = {
 			.version = VCHI_VERSION_EX(VC_SM_VER, VC_SM_MIN_VER),
 			.service_id = VC_SM_SERVER_NAME,
+			.rx_fifo_size = 0,
+			.tx_fifo_size = 0,
 			.callback = vc_sm_cma_vchi_callback,
 			.callback_param = instance,
+			.want_unaligned_bulk_rx = 0,
+			.want_unaligned_bulk_tx = 0,
+			.want_crc = 0
 		};
 
 		status = vchi_service_open(vchi_instance,
-- 
2.19.1

